### PR TITLE
Upgrade to python 3.12

### PR DESCRIPTION
**Pull request by RyanWhite25**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -8,14 +8,15 @@ readme = "README.md"
 packages = [{ include = "cruft" }]
 
 [tool.poetry.dependencies]
-python = ">=3.7"
+python = ">=3.11"
```
Comment (dannyburke1) [inline pyproject.toml]: does this need to be `3.12`?

File: pyproject.toml [hunk 2]
```diff
@@ -8,14 +8,15 @@ readme = "README.md"
...
 cookiecutter = ">=1.7"
 gitpython = ">=3.0"
 toml = { version = ">=0.10", optional = true, python = "<3.11" }
 typer = ">=0.4.0"
 click = ">= 7.1.2"
 #examples = {version = ">=1.0.2", optional = true}
 importlib-metadata = {version = ">=2.0.0,<5.0.0", python = "<3.8"}
+flake8 = "^7.1.1"
```
Comment (252afh) [inline pyproject.toml:19]: further improvement, can we use `ruff` instead of `flake8`, `isort` and `black`

Comment (252afh) [reply pyproject.toml:19]: _in another PR, not expected here_

**Review by 252afh — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (RyanWhite25) [reply pyproject.toml:19]: It's probably worth us having a discussion as a team about this & how much we want further divest from the base cruft codebase. The hope is that this fork will become redundant if/when cruft becomes actively maintained again & the features we need are merged into the base repository, so I'm conscious of sinking too much time into optimisation of the existing solution outside of what we need to get it working for our use case. Happy to have a chat about it though

File: pyproject.toml (hunk 1 above)
Comment (RyanWhite25) [reply pyproject.toml]: This is updated now. I was intending to support & run tests for both versions initially, but we enforce v3.12 of python in the cookiecutter repos, so I don't see the harm in expecting 3.12 when generating the repo

**Pull request merged**